marlin/src/marlin/marlin.rs

```diff
@@ -114,32 +110,9 @@ impl<
         let (committer_key, verifier_key) =
             PC::trim(&srs, circuit.max_degree(), supported_hiding_bound, Some(&coeff_support))?;
 
-        let mut vanishing_polys = vec![];
-        if MM::RECURSION {
```

Comment:
We no longer support recursion in the default protocol. This means that the index does not include commitments to vanishing polynomials.

This change was made because recursive provers can always compute the appropriate vanishing polynomials.